etna/search/templatetags/search_tags.py

```diff
@@ -57,14 +62,28 @@ def query_string_exclude(
 
 
 @register.filter
-def include_hidden_fields(visible_field_names, form) -> str:
+def include_hidden_fields(form_id_or_visible_field_names, form) -> str:
     """
     Returns automatically generated html hidden fields for the input form.
     The hidden fields are derived from the input form less the visible field names.
     A random suffix is applied to html id allowing the same form to be rendered
     multiple times without field ID clashes.
     """
     html = ""
+    if form_id_or_visible_field_names == "catalogue-sort-form-mobile":
+        visible_field_names = "sort_by"
+    elif form_id_or_visible_field_names == "catalogue-filters-form":
+        visible_field_names = "filter_keyword collection topic level closure opening_start_date opening_end_date created_start_date created_end_date catalogue_source held_by type country location"
+        if form.errors:
+            # visible non dynamic input fields when in error, includes hidden dynamic input fields
+            visible_field_names = "created_start_date created_end_date opening_start_date opening_end_date"
+    elif form_id_or_visible_field_names == "catalogue-search-form":
+        visible_field_names = "q"
+    elif form_id_or_visible_field_names == "catalogue-sort-form-desktop":
+        visible_field_names = "sort_by"
+    else:
+        visible_field_names = form_id_or_visible_field_names
```

Comment:
We don't want this configuration living here in this stray function that nobody ever looks at. It should be down to the calling code to specify the fields that are shown each time. Can you think of another way for this to be provided?

Reply:
Similar to comment -

> A custom filter just isn't the right device here. This should really be [simple tag](https://docs.djangoproject.com/en/4.2/howto/custom-template-tags/#simple-tags), and we need to find a way for these values to be specified on the form instance, so that they aren't hardcoded in the template

Perhaps would it be bound field methods, but I'm not sure how to go about it with the complexities for this specific page with multiple forms and their behaviours (and upcoming tickets to reflect that).

Reply:
I've hinted at this in other comments. If this is a simple tag that accepts either an `exclude` or `include` argument, you can figure out the real field list by looking at `form.fields` and subtracting from that as necessary.

Reply:
> I've hinted at this in other comments. If this is a simple tag that accepts either an exclude or include argument, you can figure out the real field list by looking at form.fields and subtracting from that as necessary.

I am aware of the form field as its implemented [here ](https://github.com/nationalarchives/ds-wagtail/blob/f17e917ada03a73c8a6062a5abaadf9156cf579d/etna/search/templatetags/search_tags.py#L88-L106) . raised a query below 

> @ababic how will this cover [here-include_hidden_fields](https://github.com/nationalarchives/ds-wagtail/blob/f17e917ada03a73c8a6062a5abaadf9156cf579d/etna/search/templatetags/search_tags.py#L93-L95) and [here-test_render_invalid_date_range_all_dates](https://github.com/nationalarchives/ds-wagtail/blob/f17e917ada03a73c8a6062a5abaadf9156cf579d/etna/search/tests/test_views.py#L641-L643)